modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java

```diff
@@ -454,63 +452,38 @@ else if (type.equals(DDMImpl.TYPE_SELECT)) {
 
 		Document document = createDocument(indexType, type, value, locale);
 
-		com.liferay.portal.kernel.search.Field field =
+		com.liferay.portal.kernel.search.Field ddmField =
 			new com.liferay.portal.kernel.search.Field("");
 
-		field.addField(
+		ddmField.addField(
 			new com.liferay.portal.kernel.search.Field(DDM_FIELD_NAME, name));
 
-		Stream.of(
-			document.getFields()
-		).map(
-			Map::entrySet
-		).flatMap(
-			Set::stream
-		).map(
-			Map.Entry::getValue
-		).forEach(
-			entry -> {
-				if (entry.getGeoLocationPoint() != null) {
-					com.liferay.portal.kernel.search.Field geolocationField =
-						new com.liferay.portal.kernel.search.Field(
-							entry.getName());
-
-					geolocationField.setGeoLocationPoint(
-						entry.getGeoLocationPoint());
-
-					field.addField(geolocationField);
-				}
-				else {
-					field.addField(
-						new com.liferay.portal.kernel.search.Field(
-							entry.getName(), entry.getValue()));
-				}
+		Map<String, com.liferay.portal.kernel.search.Field> fields =
+			document.getFields();
 
-				Stream<com.liferay.portal.kernel.search.Field> stream =
-					field.getFields(
-					).stream();
-
-				Optional<com.liferay.portal.kernel.search.Field>
-					valueFieldName = stream.filter(
-						currentField -> currentField.getName(
-						).equals(
-							"valueFieldName"
-						)
-					).findFirst();
-
-				if (!valueFieldName.isPresent() &&
-					!entry.getName().contains(
-						com.liferay.portal.kernel.search.Field.
-							SORTABLE_FIELD_SUFFIX)) {
-
-					field.addField(
-						new com.liferay.portal.kernel.search.Field(
-							"valueFieldName", entry.getName()));
-				}
+		String valueFieldName = null;
+
+		for (com.liferay.portal.kernel.search.Field field : fields.values()) {
+			ddmField.addField(field);
+
+			String fieldName = field.getName();
+
+			if ((valueFieldName == null) &&
+				!fieldName.contains(
+					com.liferay.portal.kernel.search.Field.
+						SORTABLE_FIELD_SUFFIX)) {
+
+				valueFieldName = fieldName;
 			}
-		);
+		}
+
+		if (valueFieldName != null) {
+			ddmField.addField(
+				new com.liferay.portal.kernel.search.Field(
+					DDM_VALUE_FIELD_NAME, valueFieldName));
+		}
```

Comment:
@jorgediaz-lr, is this code addressing the Geolocation field indexation properly too? I'm asking this because the removed code had a special treatment for this field.